code/libraries/tld-parser/src/main/java/nu/marginalia/util/TldParser.java

```diff
@@ -0,0 +1,83 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Parses the public suffix list into a HashSet.
+ * Provides utility functions for stripping TLDs from the domain.
+ */
+public class TldParser {
+
+    /** Parsed list of TLDs from the publicly available list */
+    public Set<String> tlds;
```

Comment:
We should probably call them domainSuffixes or something like that, since they're not strictly TLDs.